### ISSUE TITLE
can not connect to the host

### ISSUE THREAD

**Issue by DamaoShao** [paidem/guacozy#1]
I run the project with the command:
> docker run -it --rm -p 8080:80 -p 8443:443 guacozy/guacozy-server

then I add a connection, but when I connect to it,  I get the error:
> 2019-10-28 09:13:19,494 ERROR    Exception inside application: [Errno -2] Name does not resolve
  File "/usr/local/lib/python3.7/site-packages/channels/sessions.py", line 183, in __call__
    return await self.inner(receive, self.send)
  File "/usr/local/lib/python3.7/site-packages/channels/middleware.py", line 41, in coroutine_call
    await inner_instance(receive, send)
  File "/usr/local/lib/python3.7/site-packages/channels/consumer.py", line 62, in __call__
    await await_many_dispatch([receive], self.dispatch)
  File "/usr/local/lib/python3.7/site-packages/channels/utils.py", line 52, in await_many_dispatch
    await dispatch(result)
  File "/usr/local/lib/python3.7/site-packages/channels/consumer.py", line 73, in dispatch
    await handler(message)
  File "/usr/local/lib/python3.7/site-packages/channels/generic/websocket.py", line 175, in websocket_connect
    await self.connect()
  File "./guacozy_server/guacdproxy/consumers.py", line 119, in connect
    await sync_to_async(self.gclient.handshake)(**parameters, **params)
  File "/usr/local/lib/python3.7/site-packages/asgiref/sync.py", line 244, in __call__
    return await asyncio.wait_for(future, timeout=None)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 414, in wait_for
    return await futlt = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.7/site-packages/asgiref/sync.py", line 277, in thread_handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/guacamole/client.py", line 167, in handshake
    self.send_instruction(Instruction('select', protocol))
  File "/usr/local/lib/python3.7/site-packages/guacamole/client.py", line 136, in send_instruction
    return self.send(instruction.encode())
  File "/usr/local/lib/python3.7/site-packages/guacamole/client.py", line 122, in send
    self.client.sendall(data.encode())
  File "/usr/local/lib/python3.7/site-packages/guacamole/client.py", line 71, in client
    (self.host, self.port), self.timeout)
  File "/usr/local/lib/python3.7/socket.py", line 707, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "/usr/local/lib/python3.7/socket.py", line 748, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
  [Errno -2] Name does not resolve

What can I do to fix this problem？ 
Thank you！

**Comment by paidem** [comment]
Hi

 

The project is not yet really in phase for public, I still write documentation for it :) 

 

 

But I think your problem is – you don’t have a guacd server. Default guacd server is registered as “guacd:4822”, which will only be available if you run via docker-compose and have guacd container with name “guacd”

 

I just started adding documentation:

https://guacozy.readthedocs.io/en/latest/

 

Look at the architecture image on the bottom of the page. 

 

You either need to to run additional container with guacd alongside it 

docker run --name guacdlocal --restart always -d -p 4822:4822 linuxserver/guacd

 

And update Application Settings to point to new container.

 

I updated README to include this info.

 

 

But you really better use docker-compose example. It’s much easier.

 

Pavel Paidem
